src/validator/SessionKeyValidator.sol

```diff
@@ -254,6 +238,40 @@ contract SessionKeyValidator is IKernelValidator {
         }
     }
 
+    function _verifyParam(
+        address sessionKey,
+        bytes calldata callData,
+        Permission calldata _permission,
+        bytes32[] calldata _merkleProof
+    ) internal returns (ValidAfter validAfter, bool verifyFailed) {
+        SessionData storage session = sessionData[sessionKey][msg.sender];
+        bool isExecute = bytes4(callData[0:4]) == Kernel.execute.selector;
+        require(address(bytes20(callData[16:36])) == _permission.target, "SessionKeyValidator: target mismatch");
```

Comment:
I have mixed feelings on require messages, 2 key reasons why i use require instead of custom revert is 1) it's easier to develop and ship fast 2) custom error is hard to debug on when it reverted on validation phase

for 1), yes it can be improved if we have more time to build the protocol so feel free to open a PR but for 2), we do use custom errors on parts where we fully integrated the sdk which handles most of the error on the sdk side, but when we are building new validator, i prefer adding visibility on to debug

Reply:
Yeah totally agree for the initial fast dev / debug phase :)

Custom error is progressively getting easier & easier to debug with, you now have the `forge selectors up --all` that upload every custom error / function / events selectors on the [signature directory](https://www.4byte.directory/).
And now viem automatically support custom error (finding it from the sig database / abi, returning the error name, along with the raw selector: [their test](https://github.com/wagmi-dev/viem/blob/main/src/actions/public/simulateContract.test.ts#L242), [the pr mentioning it](https://github.com/wevm/viem/discussions/589))

I will push you a PR with switch to error on every file except the WeightedEcdsa & SessionKeyValidator (since a lot of modification on them currently in progress aha)

Reply:
I mean the entrypoint does not give the custom error details. Which is quite stressful atm. it only gives back the details of require statements https://github.com/eth-infinitism/account-abstraction/blob/558e024161cf355e5a0fa59b1b5c0c5cb5aeac74/contracts/core/EntryPoint.sol#L441